state/pruner.go

```diff
@@ -63,18 +63,10 @@ func NewPruner(stateStore Store, bs BlockStore, logger log.Logger, options ...Pr
 }
 
 func (p *Pruner) OnStart() error {
-	if err := p.BaseService.OnStart(); err != nil {
```

Comment:
Thanks Thane! Btw, we use this exact same code in the p2p service, is there a reason why its used there ? Or should we also refactor it there as well.

Reply:
I wouldn't refactor the P2P code yet until we're actually changing something there. We should honestly refactor this whole `Service` model entirely at some point - it's trying to simulate some kind of object-oriented way of programming that isn't at all ergonomic or idiomatic in Go.